lib/ansible/cli/inventory.py

```diff
@@ -277,6 +281,47 @@ def inventory_graph(self):
         else:
             raise AnsibleOptionsError("Pattern must be valid group name when using --graph")
 
+    def _graph_group_dot(self, group, depth=0):
+        from hashlib import sha1
+        depth = depth + 1
+        hashed_group_name = sha1(group.name.encode()).hexdigest()
+        result = ["%s\"%s\" [label=\"%s\"];" % (' ' * depth * 4, hashed_group_name, group.name)]
+        for kid in sorted(group.child_groups, key=attrgetter('name')):
+            hashed_kid_name = sha1(kid.name.encode()).hexdigest()
+            result.extend(self._graph_group_dot(kid, depth))
+            new_group = '%s"%s" -> "%s";' % (
+                ' ' * depth * 4, hashed_group_name, hashed_kid_name)
+            result.append(new_group)
+
+        if group.name != 'all':
+            for host in sorted(group.hosts, key=attrgetter('name')):
+                hashed_host_name = sha1(host.name.encode()).hexdigest()
+                new_host = '%s"%s" [label="%s"];' % (' ' * depth * 4, hashed_host_name, host.name)
+                result.append(new_host)
+                new_parent_group = '%s"%s" -> "%s";' % (
+                    ' ' * depth * 2, hashed_group_name, hashed_host_name)
+                result.append(new_parent_group)
+        return result
+
+    def inventory_graph_dot(self):
+
+        start_at = self._get_group(context.CLIARGS['pattern'])
+        if context.CLIARGS['show_vars']:
+            raise AnsibleOptionsError("This would be graphic")
```

Comment:
This should probably just be done with a mutually exclusive group in the CLI argparse declaration.

Reply:
is this what you had in mind ? cf 629dcc6bf4